Sources/Workspace/DefaultPluginScriptRunner.swift

```diff
@@ -148,15 +174,20 @@ public struct DefaultPluginScriptRunner: PluginScriptRunner {
         let executableFile = cacheDir.appending(component: "compiled-plugin")
         command += ["-o", executableFile.pathString]
         
-        // Make sure the cache directory in which we'll be placing the compiled executable exists.
-        try FileManager.default.createDirectory(at: cacheDir.asURL, withIntermediateDirectories: true, attributes: nil)
+        // Create the cache directory in which we'll be placing the compiled executable if needed. Any failure to create it will be reported by the compiler.
+        try? FileManager.default.createDirectory(at: cacheDir.asURL, withIntermediateDirectories: true, attributes: nil)
         
-        // Invoke the compiler and get back the result.
-        let compilerResult = try Process.popen(arguments: command, environment: toolchain.swiftCompilerEnvironment)
-
-        // Finally return the result. We return the path of the compiled executable only if the compilation succeeded.
-        let compiledExecutable = (compilerResult.exitStatus == .terminated(code: 0)) ? executableFile : nil
-        return PluginCompilationResult(compiledExecutable: compiledExecutable, diagnosticsFile: diagnosticsFile, compilerResult: compilerResult)
+        Process.popen(arguments: command, environment: toolchain.swiftCompilerEnvironment, queue: queue) { result in
+            switch result {
```

Comment:
result.tryMap

Reply:
Cool, thanks!  I was thinking there must be a shorthand for this pattern.